sql/sql_alter.cc

```diff
@@ -209,6 +209,35 @@ bool Alter_info::supports_lock(THD *thd, enum_alter_inplace_result result,
   return false;
 }
 
+bool Alter_info::vers_prohibited(THD *thd) const
+{
+  if (thd->slave_thread ||
+      thd->variables.vers_alter_history != VERS_ALTER_HISTORY_ERROR)
+  {
+    return false;
+  }
+  if (flags & (
+    ALTER_PARSER_ADD_COLUMN |
+    ALTER_PARSER_DROP_COLUMN |
+    ALTER_CHANGE_COLUMN |
+    ALTER_COLUMN_ORDER))
+  {
+    return true;
+  }
+  if (flags & ALTER_ADD_INDEX)
```

Comment:
There are also
```
#define ALTER_ADD_UNIQUE_INDEX               (1ULL << 38)
#define ALTER_ADD_PK_INDEX                   (1ULL << 40)
```
perhaps you can check that instead of `ALTER_ADD_INDEX` ?

Reply:
Good point. This may save some iterations.